language/stdlib/modules/LibraAccount.move

```diff
@@ -1217,20 +1297,30 @@ module LibraAccount {
             TransactionFee::pay_fee(Libra::withdraw(coin, transaction_fee_amount))
         }
     }
+
     spec fun epilogue {
         /// > TODO: timeout
         pragma verify = false;
     }
 
-    /// Bump the sequence number of an account. This function should be used only for bumping the sequence number when
-    /// a writeset transaction is committed.
-    fun bump_sequence_number(signer: &signer) acquires LibraAccount {
-        let addr = Signer::address_of(signer);
-        assert(exists_at(addr), Errors::not_published(EACCOUNT));
-        let sender_account = borrow_global_mut<LibraAccount>(addr);
-        sender_account.sequence_number = sender_account.sequence_number + 1;
+    /// Epilogue for WriteSet trasnaction
+    fun writeset_epilogue(
+        lr_account: &signer,
+        writeset_payload: vector<u8>,
+        txn_sequence_number: u64
+    ) acquires LibraWriteSetManager, LibraAccount, Balance {
+        let writeset_events_ref = borrow_global_mut<LibraWriteSetManager>(CoreAddresses::LIBRA_ROOT_ADDRESS());
+
+        Event::emit_event<UpgradeEvent>(
+            &mut writeset_events_ref.upgrade_events,
+            UpgradeEvent { writeset_payload },
+        );
+        // Currency code don't matter here as it won't be charged anyway.
+        epilogue<LBR::LBR>(lr_account, txn_sequence_number, 0, 0, 0);
```

Comment:
I'm a bit torn over here. On one hand I love this idea of having separate functions we can call to get the logics we want. On the other hand, I'm trying to minimize the code change and keep the code structure as it was (and try to merge it ASAP as soon as #5997 lands to leverage the fact that it is touching the genesis init order). I'm also wasn't sure how it might affect the prover side work. Do you have any strong preference over here? cc @tzakian @sblackshear @shazqadeer  for thoughts as well.

Reply:
I feel like splitting it up might make the prover's life easier? beats me though
If its seems like work to convince a human that the checks aren't happening, I feel like it might be cumbersome to convey this in the spec as well, but I could be wrong

Reply:
I don't think splitting it would make the prover's life easier (but, it's one of those things where it's hard to determine exactly what will make it harder/easier until you start proving it). I think for now in the interest of getting the breaking change in for this release cycle I would like to hold off on this, and if we do decide to refactor it out we can later on and that shouldn't be a breaking change.